### PR TITLE
Kafka topic name sanitizer

### PR DESCRIPTION
**Pull request by joao-conde**
| - | - |
| --- | --- |
| Issue | -- |
| Dependencies | -- |
| Decisions | Adapt on the in/out the kafka topic name. Kafka topics have name [restrictions](https://github.com/apache/kafka/blob/0.10.2/core/src/main/scala/kafka/common/Topic.scala#L29). We already use the adapter pattern so this change does not structurally change anything. |

### CODE REVIEW

File: js/kafka/kafka-consumer.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { conf } from "yonius";
 import { KafkaClient } from "./kafka-client";
+import { sanitizeTopicName } from "./utils";
 import { Consumer } from "../consumer";
-
```
Comment (gcandal) [inline js/kafka/kafka-consumer.js:4]: Revert, keep the newline.

File: js/kafka/utils.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export const sanitizeTopicName = topic => {
```
Comment (gcandal) [inline js/kafka/utils.js]: Move to the existing file (https://github.com/ripe-tech/ripe-bus-api-js/blob/master/js/kafka/kafka-client.js), use the same way to declare a function (using `function()...` and add jsdocs.

**Review by gcandal — CHANGES_REQUESTED** [review]

**Comment by joamag** [comment]
@joao-conde please re-check that everything is still working, another major design change happened

**Comment by joamag** [comment]
Adding fast-track to try to merge this ASAP

**Comment by joao-conde** [comment]
Everything worked and passed the following manual tests:

```js
// bind with event filtering
const callback = message => console.log(message);
(async () => {
    try {
        await bus.bind("topic", { 
            events: ["topic.created", "topic.sent", "topic.tobias"],
            callback: callback,
        })
    } catch (e) {}
})();

// trigger with unsanitized topic name
(async () => {
    try {
        await bus.trigger("ripe_twitch:order.created", "Unsanitized topic name");
    } catch (e) {}
})();

// trigger with metadata override
(async () => {
    try {
        await bus.trigger("topic.created", "topic.created", {
            origin: "overriden",
            hostname: "overriden",
            datatype: "json",
            timestamp: 10000
        });
    } catch (e) {}
})();

// trigger with topic override
(async () => {
    try {
        await bus.trigger("another.created", "Hello World", {
            topic: "topic"
        });
    } catch (e) {}
})();
```

**Review by joamag — APPROVED** [review]

**Comment by joamag** [comment]
Releasing new version now!

**Comment by joamag** [comment]
0.3.0

**Pull request merged**